### PR TITLE
fix(CI): Add migrations integration tests

### PR DESCRIPTION
**Pull request by ahmed-nour-fdc**

### CODE REVIEW

File: database/migrations/postgres/1721378995777505_rename_application_locks.up.sql [hunk 1]
```diff
@@ -1,2 +1,9 @@
-ALTER TABLE IF EXISTS application_locks
```
Comment (AminSlk) [inline database/migrations/postgres/1721378995777505_rename_application_locks.up.sql:1]: Doesn't the `IF EXISTS` part here work?

File: database/migrations/postgres/1720012750850077_rename_commit_events.up.sql [hunk 2]
```diff
@@ -1,2 +1,6 @@
-ALTER TABLE IF EXISTS events
-    RENAME TO commit_events;
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = 'commit_events') THEN
+        ALTER TABLE events RENAME TO commit_events;
```
Comment (AminSlk) [inline database/migrations/postgres/1720012750850077_rename_commit_events.up.sql:4]: I think we should still keep the `IF EXISTS` part after the ALTER TABLE

File: database/migrations/postgres/1721378995777505_rename_application_locks.up.sql [hunk 3]
```diff
@@ -1,2 +1,9 @@
...
+DO $$ BEGIN IF NOT EXISTS (
+    SELECT 1
+    FROM information_schema.tables
+    WHERE table_name = 'app_locks'
+) THEN
+ALTER TABLE application_locks
```
Comment (AminSlk) [inline database/migrations/postgres/1721378995777505_rename_application_locks.up.sql:6]: Same as above

File: tests/integration-tests/migrations_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		t.Fatalf("Error pinging database: %v", err)
+	}
+	if err := deleteSchemaMigrationsTable(dbConfig); err != nil {
+		t.Fatalf("Failed to delete schema migrations table: %v", err)
+	}
+	testCases := []struct {
+		name        string
+		postRunFunc func(db.DBConfig) error
+	}{
+		{
```
Comment (sven-urbanski-freiheit-com) [inline tests/integration-tests/migrations_test.go:61]: these 2 elements don't seem to be independent test cases.
So I would suggest to have just 1 element here, and in the actual test (line ~73), to run `RunDBMigrations` twice.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: database/migrations/postgres/1720012750850077_rename_commit_events.up.sql (hunk 2 above)
Comment (ahmed-nour-fdc) [reply database/migrations/postgres/1720012750850077_rename_commit_events.up.sql]: Done

**Pull request merged**